File: care_well_backend/CareWell/CareWell.Domain/Salud/FichaSaludAntecedente.cs
using CareWell.Domain.ValueObjects.Salud;
using CareWell.Global.Exceptions;
using CareWell.Global.Mensajes;

namespace CareWell.Domain.Salud
{
    public class FichaSaludAntecedente : BaseEntity
    {
        public virtual FichaSalud FichaSalud { get; private set; }

        public virtual string Nombre { get; private set; }

        public virtual string Descripcion { get; private set; }

        public virtual string VinculoFamiliar { get; private set; }

        public virtual void Registrar(AgregarAntecedente agregarAntecedente, FichaSalud fichaSalud)
        {
            if (string.IsNullOrEmpty(agregarAntecedente.Nombre))
                throw new ValidacionDominioException(Mensajes.NombreAntecedenteRequerido);

            if (string.IsNullOrEmpty(agregarAntecedente.Descripcion))
                throw new ValidacionDominioException(Mensajes.DescripcionAntecedenteRequerido);

            if (string.IsNullOrEmpty(agregarAntecedente.VinculoFamiliar))
                throw new ValidacionDominioException(Mensajes.VinculoAntecedenteRequerido);

            this.FichaSalud = fichaSalud;
            this.Nombre = agregarAntecedente.Nombre;
            this.Descripcion = agregarAntecedente.Descripcion;
            this.VinculoFamiliar = agregarAntecedente.VinculoFamiliar;
        }
    }
}

File: care_well_app/lib/domain/entities/health/ficha_salud_antecedente.dart
import 'package:care_well_app/domain/entities/entities.dart';

class FichaSaludAntecedente extends BaseEntity {
  final String nombre;
  final String descripcion;
  final String vinculoFamiliar;

  const FichaSaludAntecedente({
    super.id = 0,
    required this.nombre,
    required this.descripcion,
    required this.vinculoFamiliar,
  });

  @override
  FichaSaludAntecedente copyWith({
    int? id,
    String? nombre,
    String? descripcion,
    String? vinculoFamiliar,
  }) {
    return FichaSaludAntecedente(
      id: id ?? this.id,
      nombre: nombre ?? this.nombre,
      descripcion: descripcion ?? this.descripcion,
      vinculoFamiliar: vinculoFamiliar ?? this.vinculoFamiliar,
    );
  }
}
